features/step_definitions/common_steps.rb

```diff
@@ -483,7 +483,12 @@ def enable_pool(target)
 end
 
 When(/^I register using "([^"]*)" key$/) do |arg1|
-  regurl = "http://#{$server_ip}/XMLRPC"
+  # register to the proxy, or if none, to the server
+  if $proxy_ip == 'None'
+    regurl = "http://#{$server_ip}/XMLRPC"
+  else
+    regurl = "http://#{$proxy_ip}/XMLRPC"
```

Comment:
dont use $proxy_ip here.
we should introduce $proxy.ip .

If you add a method in this module, we can have it
https://github.com/SUSE/spacewalk-testsuite-base/blob/master/features/support/lavanda.rb#L7

Reply:
with this we can remove the global variable

Reply:
I suggest we keep refactoring work like this separate from this PR.